### ISSUE TITLE
region attachHtml default when template: false

### ISSUE THREAD

**Issue by spenoir** [marionettejs/backbone.marionette#2587]
When `template: false` on a view that you attach using a Region, it doesn't make sense to `this.el.appendChild(view.el);`. I know I can just override it but why not just make it the default like this:
      
      ...
      // Override this method to change how the new view is
      // appended to the `$el` that the region is managing
      attachHtml: function(view) {
        if (view.template !== false) {
          this.$el.contents().detach();
          this.el.appendChild(view.el);
        }
      },
      ...

Let me know if you want me to do a pull request.
I reckon all occurrences of attachHtml should include the template false conditional actually. Thanks

**Comment by ianmstew** [comment]
While I see where you're going with this, a view with no template still has a single element that could potentially be significant (button, hyperlink, form--see `tagName`, `className`, `id`, `attributes` in Backbone doc), and we still need DOM attachment to ensure expected events like `attach` to fire.

What's your goal for not appending a template-less view's element to the DOM?  Is it avoiding a `div` inside the region that isn't doing much?  If so, there's an effort underway to eliminating redundancy between a Region's el and the view's el it contains (#2625).

**Comment by spenoir** [comment]
*What's your goal for not appending a template-less view's element to the DOM?* To avoid replacing the html onRender when the template equals false. 
If an empty div gets added on every render then the template is not truly false in value. Would the other issue fix this by not adding an empty div? That sounds like it would fix this issue.

Each time the view renders, its el value should be taken from the DOM, regardless of whether its in a region or not I suppose; only when template equals false though of cause.

**Comment by ianmstew** [comment]
@ahumphreys87, What would happen when a view with `template: false` replaces a region's el?  Does it re-use the region's element?

**Comment by ahumphreys87** [comment]
Im not sure the tests cover that case at the moment.

I think it should just replace it with the empty view tag - if template is `false` you still get that right?

**Comment by paulfalgout** [comment]
Hmm  I find `template:false` to be most useful for single element items as @ianmstew mentioned.   If a view doesn't have a template, it doesn't mean it doesn't have an `el` that needs rendering.  This issue seems specific to showing in a region and we may need to figure out how to handle it their, or in the least suggest a simple cookbook solution.

**Comment by spenoir** [comment]
Yes I see your point Paul. I have a workaround where I override attachHtml
when using a ChildView but it could probably be used in a similar way for
any View, I've got it in a gist here:
https://gist.github.com/spenoir/0edca0ec8bb94c72aefc

**Comment by spenoir** [comment]
I've updated that gist btw

**Comment by Tranfu** [comment]
this.el.appendChild(view.el); 
IE8 Error: Unexpected call to method or property
how to fix it?!

**Comment by paulfalgout** [comment]
Apologies for not following up on this earlier.  I think in v3 with the work around existing DOM the thought now is that if the view isn't rendering, then `onRender` isn't the correct hook for re-setting up stuff.  So I don't think this is an optional solution.  We can reopen if it seems necessary.
